File: support/countdb.pl
#!/usr/bin/env perl

###
## @(#) countdb		falcon@icb.net
## Updated by hoche@grok.com       5/24/19
##
## count how many nicks there are. though effective, this is slow & inefficient
###

$num = 0;

dbmopen(my %DB, "./icbdb", 0666);
while (my ($key, $val) = each(%DB) )
{
    if ( $key =~ /^(.+).nick$/ )
    {
        $num++;
    }
}

printf("%d nicknames listed.\n", $num);

dbmclose(%DB);

File: support/countdb.py
#!/usr/bin/env python3

###
## countdb.py    falcon@icb.net
##
## count how many nicks there are
###

import sys
import os

sys.path.insert(0, os.path.dirname(os.path.abspath(__file__)))
from icbdb import IcbDb

def main():
    dbpath = sys.argv[1] if len(sys.argv) > 1 else "./icbdb"
    with IcbDb(dbpath) as db:
        num = sum(1 for key in db if key.endswith(".nick"))
    print(f"{num} nicknames listed.")

if __name__ == "__main__":
    main()
